src/Optimizer/Transformer/OptimizeHeroImages.php

```diff
@@ -215,6 +222,17 @@ private function findHeroImages(Document $document, $maxHeroImageCount)
                 $seenParagraphCount++;
             }
 
+            if ($node->tagName === Extension::IMG && ! $node->hasAttribute(Attribute::DATA_HERO)) {
+                $highFetchpriorityImage = $document->xpath->query(
+                    self::NOSCRIPT_IMG_FETCHPRIORITY_HIGH_XPATH_QUERY,
+                    $node
+                )->item(0);
+
+                if ($highFetchpriorityImage instanceof Element) {
+                    $node->setAttribute(Attribute::DATA_HERO, '');
```

Comment:
This creates a boolean attribute without any value:

```suggestion
                    $node->appendChild($document->createAttribute(Attribute::DATA_HERO));
```